### DISCUSSION TITLE
[General Question]

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1212] [General]
How do I do this
```F#
let detector x =
    match x with
    | :? string as x -> 0
    | :? int as x -> 1
```
I don't know what x is, but I want to pattern match it then process them separately

**Comment by bartelink** [comment]
I'd recommend closing this and using the F# slack or discord for this sort of question - this repo is a relatively low traffic repo that's intended for well-formed language proposals and/or suggestions for specific things that you have been unable to resolve via other avenues, and not so much a general F# usage Q/A forum/chat.

That's not to say there arent highly capable and helpful people that watch this repo, but please respect that it would dilute the value of this repo if this became the traffic that goes through it's notifications.

In some cases, Stack Overflow can be good for this too, but
a) it's not well enough formed for there as it is
b) the more interactive nature of discord and slack will likely get your question clarified well quickly

Bottom line is any of the other sources will have far more people
a) than the 79 watching this
b) who will be likely to to/fro with you to clarify your ask

**Reply by vzarytovskii** [reply]
It's fine to use this repo too, but you'll definitely have better luck getting answers on SO, discussions in dotnet/fsharp and discord/slack
